infoschema/cache_test.go

```diff
@@ -116,16 +116,46 @@ func TestGetLatest(t *testing.T) {
 	require.Nil(t, ic.GetLatest())
 
 	is1 := infoschema.MockInfoSchemaWithSchemaVer(nil, 1)
-	ic.Insert(is1, 1)
+	ic.Insert(is1, 1, 0)
 	require.Equal(t, is1, ic.GetLatest())
 
 	// newer change the newest
 	is2 := infoschema.MockInfoSchemaWithSchemaVer(nil, 2)
-	ic.Insert(is2, 2)
+	ic.Insert(is2, 2, 0)
 	require.Equal(t, is2, ic.GetLatest())
 
 	// older schema doesn't change the newest
 	is0 := infoschema.MockInfoSchemaWithSchemaVer(nil, 0)
-	ic.Insert(is0, 0)
+	ic.Insert(is0, 0, 0)
 	require.Equal(t, is2, ic.GetLatest())
 }
+
+func TestGetByTimestamp(t *testing.T) {
+	ic := infoschema.NewCache(16)
+	require.NotNil(t, ic)
+	require.Nil(t, ic.GetLatest())
+
+	is1 := infoschema.MockInfoSchemaWithSchemaVer(nil, 1)
+	ic.Insert(is1, 2, 1)
```

Comment:
It is impossible that a schema is updated at `t2`, but commited diff at `t1`. It suggest remove `snapshotTS` completely as suggested.